js/blocks/controls/rich-text.js

```diff
@@ -0,0 +1,120 @@
+const { BaseControl, Fill } = wp.components;
+const { RichText } = wp.editor;
+const { applyFormat, registerFormatType, removeFormat } = wp.richText;
+const { __ } = wp.i18n;
+const { AlignmentToolbar } = wp.editor;
+const ALIGNMENTS = [ 'left', 'center', 'right' ];
+const ALIGNMENT_CONTROL_NAME = 'block-lab/rich-text-alignment';
+const FORMATTING_CONTROLS = [ ALIGNMENT_CONTROL_NAME, 'bold', 'italic', 'strikethrough', 'link' ];
+
+/**
+ * Gets the styling for a given alignment.
+ *
+ * @pram {String} alignment The alignment, like 'left' or 'right'.
+ * @return {String} The alignment style, like text-align: left;.
+ */
+const getAlignmentStyle = ( alignment ) => {
+	return `text-align: ${ alignment };`
+}
+
+/**
+ * Gets the alignment type from the properties.
+ *
+ * @param {Object} alignmentProps The properties for the alignment.
+ * @return {String|null} The alignment, either 'left', 'center', or 'right'.
+ */
+const getAlignmentFromProps = ( alignmentProps ) => {
+	if ( ! alignmentProps.activeAttributes || ! alignmentProps.activeAttributes.align ) {
+		return;
+	}
+
+	const alignmentStyle = alignmentProps.activeAttributes.align;
+	const matchedAlignments = ALIGNMENTS.filter( ( possibleAlignment ) => {
+		return getAlignmentStyle( possibleAlignment ) === alignmentStyle;
+	} );
+
+	return matchedAlignments.length ? matchedAlignments[ 0 ] : null;
+};
+
+registerFormatType(
+	ALIGNMENT_CONTROL_NAME,
+	{
+		title: __( 'Alignment Controls', 'block-lab' ),
+		tagName: 'div',
+		className: 'bl-aligned',
+		attributes: {
+			align: 'style',
+		},
+		edit: ( props ) => {
+			const fillName = `RichText.ToolbarControls.${ ALIGNMENT_CONTROL_NAME }`;
+			const value = getAlignmentFromProps( props );
+
+			return (
+				<Fill name={ fillName }>
+					<AlignmentToolbar
+						value={ value }
+						onChange={ ( newAlignment ) => {
+							if ( newAlignment ) {
+								props.onChange(
+									applyFormat(
+										props.value,
+										{
+											type: ALIGNMENT_CONTROL_NAME,
+											attributes: {
+												align: getAlignmentStyle( newAlignment ),
```

Comment:
Heavy nesting… but I'm okay with it.

Reply:
You could potentially spin out the `onChange` handler into a separate method. But no real need.

Reply:
Yeah, good point.

Reply:
https://github.com/getblocklab/block-lab/pull/221/commits/4afb3f90c1d420195ed2980c77d20671ed812306#diff-b64eb5a514d6a8f70b18938222cd7a4fR45 applies your suggestion of creating a separate method for this.